students/BrandonHenson/Lesson9/Mailroom_5.py

```diff
@@ -0,0 +1,151 @@
+            fileobj.close()
+
+    def menu_2(self):
+        try:
+            print(f'{"Donor Name":18s} {"|  Total Given":12s} \
+    {"|  Num Gifts  |":14s} {"Average Gift":18s}')
+            print(f'{"-"*62}')
+            [print(f'{name:18s} ${sum(self.donors[name]):10.2f} {len(self.donors[name]):8d}        \
+    ${sum(self.donors[name])/len(self.donors[name]):10.2f}')
+    for name in self.donors.keys()]
+        except ZeroDivisionError:
+            pass
+
+
+def menu_1(donorhistory):
+    while True:
+        name = input("Enter a full name or 'list' to view all\n")
+        if name.lower() == 'list':
+            print(donorhistory.donors.keys())
+        else:
+            break
+
+    success = False
+    while not success:
+        donation = int(input("Donation amount? \n"))
+        try:
+            donorhistory.donate(name, donation)
+            success = True
+        except Exception as e:
+            print(e.args[0])
+            success = False
+    print("Dear {},\nThank you for your generous donation\
+ of ${}\nIt will be put to good use.\n\n Sincerely,\n\
+                -The Team".format(name, donation))
+
+
+def report(donorhistory):
+    donorhistory.menu_2()
+
+
+def menu_1_all(donorhistory):
+    [donorhistory.createfile(name) for name in donorhistory.donors.keys()]
+
+if __name__ == '__main__':
+    donor_history = {}
+    donor_history['Brandon Henson'] = [1005.49, 3116.72, 5200]
+    donor_history['Alicia Henson'] = [21.47, 1500]
+    donor_history['Michael Green'] = [2400.54]
+    donor_history['Brandon Henson Jr'] = [355.42, 579.31]
+    donor_history['Kaiya Henson'] = [636.9, 850.13, 125.23]
+
+    DonorDict = DonorAll(donor_history)
+
+    response = input('\nSelect an option:\n'
+             '[1] Send a Thank You\n'
+             '[2] Create a Report\n'
+             '[3] Send letters to everyone\n'
+             '[4] Exit\n')
+    try:
+        int(response)
+    except:
+        while not response.isdigit():
+            print("Enter 1,2,3, or 4.")
+            print()
+            response = input('\nSelect an option:\n'
+                             '[1] Send a Thank You\n'
+                             '[2] Create a Report\n'
+                             '[3] Send letters to everyone\n'
+                             '[4] Exit\n')
+    arg_dict = {1: menu_1, 2: report, 3: menu_1_all}
+    while int(response) != 4:
+        try:
+            arg_dict.get(int(response))(DonorDict)
+        except (TypeError, ValueError):
+            print('Enter 1,2,3, or 4')
+            print()
+            response = input('\nSelect an option:\n'
+             '[1] Send a Thank You\n'
+             '[2] Create a Report\n'
+             '[3] Send letters to everyone\n'
+             '[4] Exit\n')
+        else:
+            response = input('\nSelect an option:\n'
+             '[1] Send a Thank You\n'
+             '[2] Create a Report\n'
+             '[3] Send letters to everyone\n'
+             '[4] Exit\n')
```

Comment:
you can combine lines 118 to 151 to simplify and reduce duplicated code:
```
while True:
    response = input('>...')
    try:
        args_dict[int(response)](donor_dict)
    except:
        print("some error message")
```

I think this will satisfy all of your needs in those lines.